### DISCUSSION TITLE
Учебный курс: порядок следования полей в таблице

### DISCUSSION THREAD

**Discussion by d-nikolaev** [totumonline/totum-ru-issues-and-discussions#151] [Неопределенное]
В конце самого урока возник вопрос вот по этой части: 
[https://www.youtube.com/clip/UgkxKphxuGOw8ZeMUmAypLPk-EICEhKSZI9K](url)
Вы говорите h_field_12 больше h_field_11. Больше по какому значению? 
А если у полей name будут такими: stoimost и nadbavka то что здесь больше чего?

**Comment by erpic-ru** [comment]
Name поля значения не имеет. Речь идет про Sort (Порядок):
<img width="311" alt="Снимок экрана 2022-06-06 в 10 54 36" src="https://user-images.githubusercontent.com/91674114/172119641-9a623fd9-2d9e-4f0b-b6df-b7b24d877c98.png">
Это красные цифры над заголовком поля
 
Суть в том, что поле с сортом 10 не "видит" поля с сортом 11 и далее (и, соответственно, не может использовать их значения). Следует обращать внимание на этот момент в расчетах кодом.
Возможно, есть более понятное объяснение.

P.S. Про нюансы последовательности расчетов можно посмотреть тут: https://ru.docs.totum.online/calculation-sequence

**Reply by ttm-alexeypolunin** [reply]
При одинаковом `sort` (что для тотум является не лучшей практикой — надо стараться задавать разный `sort`) с 3 версии тотум будет определяет! порядок поля по `name`, сортируя их как строку.

Те `nadbavka` будет раньше чем `stoimost`

В видео наверное не самый крутой пример, тк в природе конешн существует порядок сортировки в котором цифры в конце строки обрабатываются как числа. Но тотум обрабатывает `name` целиком как строку.

**Comment by d-nikolaev** [comment]
Теперь понял. Вернее я изначально это понял по видео. Но "в моменте" на видео вы явно мышкой указывали на поля name и я со своей стороны буквально это воспринял и это меня смутило. Поэтому такой и вопрос. 
Но теперь всё ясно.
